pkg/controller/atlasproject/encryption_at_rest.go

```diff
@@ -34,6 +45,94 @@ func ensureEncryptionAtRest(ctx *workflow.Context, projectID string, project *md
 	return workflow.OK()
 }
 
+func readEncryptionAtRestSecrets(kubeClient client.Client, service *workflow.Context, encRest *mdbv1.EncryptionAtRest, parentNs string) error {
+	if encRest == nil {
+		return nil
+	}
+
+	if encRest.AwsKms.Enabled != nil || *encRest.AwsKms.Enabled || encRest.AwsKms.SecretRef.Name != "" {
+		fieldData, watchObj, err := readSecretData(kubeClient, encRest.AwsKms.SecretRef, parentNs, "AccessKeyID", "SecretAccessKey", "CustomerMasterKeyID", "Region", "RoleID")
+		service.AddResourcesToWatch(*watchObj)
+		if err != nil {
+			return err
+		}
+
+		fillStructFields(fieldData, &encRest.AwsKms)
```

Comment:
nitpick
I would create a function per cloud secret reads, will make this function shorter, higher level and easier to read.

Eg.
```suggestion
		err = readAndSetAWSSecrets(&encRest.AwsKms, kubeClient, encRest.AwsKms.SecretRef, parentNs)
```

Reply:
Good point, fixed